### PR TITLE
added solution hw-24-1

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,5 +8,69 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+
```
Comment (Nokafan) [inline pom.xml]: Redundant line

File: src/main/java/com/internet/shop/model/Order.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.List;
+
+public class Order {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public Order(List<Product> products, Long userId) {
+        this.products = products;
```
Comment (Nokafan) [inline src/main/java/com/internet/shop/model/Order.java]: Need to initialise by new ArrayList<>()

File: src/main/java/com/internet/shop/model/ShoppingCart.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.List;
+
+public class ShoppingCart {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public ShoppingCart(List<Product> products, Long userId) {
+        this.products = products;
```
Comment (Nokafan) [inline src/main/java/com/internet/shop/model/ShoppingCart.java]: Same for this list

File: pom.xml [hunk 4]
```diff
@@ -8,5 +8,69 @@
...
-</project>
\ No newline at end of file
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
Comment (KseniiaMakarova) [inline pom.xml]: remove this - we don't use any dependencies yet

File: src/main/java/com/internet/shop/dao/ProductDao.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+public interface ProductDao {
+    Product create(Product product);
+
+    Optional<Product> getById(Long productId);
+
+    Product update(Product product);
+
+    boolean deleteById(Long productId);
+
+    boolean delete(Product product);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/dao/ProductDao.java]: we can remove this method - we won't use it in our app

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        return Storage.products.stream()
+                .filter(product -> product.getId().equals(productId))
+                .findFirst();
+    }
+
+    @Override
+    public Product update(Product product) {
+        Storage.products.stream()
+                .filter(prod -> prod.getId().equals(product.getId()))
+                .forEach(prod -> prod = product);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: this doesn't cause changes in our storage
the elements are changed only in this stream

let's create IntStream of indices, than filter out only those indices on which there are elements with the same id
then for each of the remaining indices, replace the element in storage with our `product`

File: src/main/java/com/internet/shop/model/Order.java (hunk 2 above)
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/Order.java]: we won't receive `products` from outside

File: src/main/java/com/internet/shop/model/Order.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        this.userId = userId;
+    }
+
+    public List<Product> getProducts() {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/Order.java:24]: add getter and setter for id

File: src/main/java/com/internet/shop/model/ShoppingCart.java (hunk 3 above)
Comment (KseniiaMakarova) [reply src/main/java/com/internet/shop/model/ShoppingCart.java]: I agree

File: src/main/java/com/internet/shop/model/ShoppingCart.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        this.userId = userId;
+    }
+
+    public List<Product> getProducts() {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/ShoppingCart.java:24]: add getter and setter for id

File: src/main/java/com/internet/shop/model/User.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    private String login;
+    private String password;
+
+    public User(String name, String login, String password) {
+        this.name = name;
+        this.login = login;
+        this.password = password;
+    }
+
+    public String getName() {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/model/User.java:23]: same

File: src/main/java/com/internet/shop/service/ProductService.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+public interface ProductService {
+    Product create(Product product);
+
+    Product getById(Long productId);
+
+    Product update(Product product);
+
+    boolean deleteById(Long productId);
+
+    boolean delete(Product product);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/service/ProductService.java]: we won't need this, let's remove it

File: src/main/java/com/internet/shop/service/impl/ProductServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    private ProductDao productDao;
+
+    @Override
+    public Product create(Product product) {
+        return productDao.create(product);
+    }
+
+    @Override
+    public Product getById(Long productId) {
+        return productDao.getById(productId).orElseThrow();
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/service/impl/ProductServiceImpl.java]:
```suggestion
        return productDao.getById(productId).get();
```
let's use `get()` by now

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (TimoxaV) [reply pom.xml]: done

File: src/main/java/com/internet/shop/model/Order.java (hunk 2 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/model/Order.java]: Done. Thanks

File: src/main/java/com/internet/shop/model/ShoppingCart.java (hunk 3 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/model/ShoppingCart.java]: done

File: pom.xml (hunk 4 above)
Comment (TimoxaV) [reply pom.xml]: done

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 6 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: fixed

File: src/main/java/com/internet/shop/model/Order.java (hunk 2 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/model/Order.java]: fixed

File: src/main/java/com/internet/shop/model/Order.java (hunk 7 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/model/Order.java:24]: done

File: src/main/java/com/internet/shop/model/ShoppingCart.java (hunk 8 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/model/ShoppingCart.java:24]: done

File: src/main/java/com/internet/shop/model/User.java (hunk 9 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/model/User.java:23]: done

File: src/main/java/com/internet/shop/service/ProductService.java (hunk 10 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/service/ProductService.java]: done

File: src/main/java/com/internet/shop/service/impl/ProductServiceImpl.java (hunk 11 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/service/impl/ProductServiceImpl.java]: done

File: src/main/java/com/internet/shop/dao/ProductDao.java (hunk 5 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/dao/ProductDao.java]: done

File: pom.xml [hunk 12]
```diff
@@ -8,5 +8,61 @@
...
+        <jdk.version>11</jdk.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
+        <maven.checkstyle.plugin.configLocation>
+            https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
+        </maven.checkstyle.plugin.configLocation>
+    </properties>
+    <dependencies>
+    </dependencies>
```
Comment (KseniiaMakarova) [inline pom.xml]: you can remove this

**Review by KseniiaMakarova — APPROVED** [review]

File: pom.xml (hunk 12 above)
Comment (TimoxaV) [reply pom.xml]: deleted dependencies

**Pull request merged**